### ISSUE TITLE
Error on docs build for `load_connection_info`

### ISSUE THREAD

**Issue by willingc** [jupyter/jupyter_client#194]
`load_connection_info` has been added in 4.4. See changelog `.KernelClient.load_connection_info`

Documentation build of `client.rst` is failing on:

```
sers/carol/code/jupyter-projects/jupyter_client/docs/api/client.rst:21: WARNING: autodoc: failed to import method 'KernelClient.load_connection_info' from module 'jupyter_client'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/carol/anaconda/envs/jupyter_client/lib/python3.5/site-packages/sphinx/util/inspect.py", line 109, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'KernelClient' has no attribute 'load_connection_info'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/carol/anaconda/envs/jupyter_client/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 526, in import_object
    obj = self.get_attr(obj, part)
  File "/Users/carol/anaconda/envs/jupyter_client/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 422, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/Users/carol/anaconda/envs/jupyter_client/lib/python3.5/site-packages/sphinx/util/inspect.py", line 125, in safe_getattr
    raise AttributeError(name)
AttributeError: load_connection_info
```

**Comment by willingc** [comment]
@takluyver @minrk I've spent way too long trying to troubleshoot this. Which likely means that I am missing something obvious. Thoughts?

**Comment by michaelpacer** [comment]
So If you look at https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/client.py#L35 it does not have a method called `load_connection_info`, but the docs/api/client.rst file expects it

.. autoclass:: KernelClient

   .. automethod:: load_connection_file

   .. automethod:: load_connection_info

   .. automethod:: start_channels

   .. automethod:: execute

   .. automethod:: complete

   .. automethod:: inspect

   .. automethod:: history

   .. automethod:: comm_info

   .. automethod:: is_complete

   .. automethod:: input

   .. automethod:: shutdown

   .. automethod:: get_shell_msg

   .. automethod:: get_iopub_msg

   .. automethod:: get_stdin_msg

So just delete  .. automethod:: load_connection_info and it should stop erring.

**Comment by michaelpacer** [comment]
Ah — correction… it's just not looking in the right place. It's looking in `client.py` when it should be looking at `connect.py` https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/connect.py#L431 if it wants to find that method.

**Comment by michaelpacer** [comment]
You want to be calling it as an automethod from `ConnectionFileMixin` in `connect.rst` not `KernelClient`.

**Comment by willingc** [comment]
@michaelpacer That's true. Yet, load_connection_file does build without erring. Wondering why this one works and the other doesn't.

**Comment by willingc** [comment]
Earlier this evening, I did delete load_connection_info and all was fine. Was trying to figure out why one worked and the other did not with the ConnectionFileMixin. I suppose that the version being mixed in is not the latest perhaps. I'm going to take a break and pick this up in the morning ;-) Good luck on the dis. work.

**Comment by michaelpacer** [comment]
It's likely because load_connecton_file has an argument with a default value whereas the other needs to be defined with an explicit secondary argument, thus though it inherits from  ConnectionFileMixin it doesn't populate that argument and is never actually declared as a method?

**Comment by minrk** [comment]
When I build the docs locally, load_connection_info, which is an inherited method on JupyterClient, does render correctly. If it's failing, that suggests that the jupyter_client being imported to run autodoc is not actually the version in the repo, but probably the most recent stable release.

We need to make sure that `python.pip_install = true` in the readthedocs config in order for the repo version of the package to be the one that is installed. This should be fixed by #196.
